examples/http/tracer_common.hpp

```diff
@@ -13,8 +13,10 @@ void initTracer() {
       new opentelemetry::exporter::trace::OStreamSpanExporter);
   auto processor = std::unique_ptr<sdktrace::SpanProcessor>(
       new sdktrace::SimpleSpanProcessor(std::move(exporter)));
+  std::vector<std::unique_ptr<sdktrace::SpanProcessor>> processors;
+  processors.push_back(std::move(processor));
   // Default is an always-on sampler.
-  auto context = std::make_shared<sdktrace::TracerContext>(std::move(processor));
+  auto context = std::make_shared<sdktrace::TracerContext>(std::move(processors));
```

Comment:
Should there be a convenience constructor that just takes on processor? 

Alternatively, a static helper method to make TracerProvider from a processor?

Reply:
+1.

This is what OTel.NET does https://github.com/open-telemetry/opentelemetry-dotnet/blob/a25741030f05c60c85be102ce7c33f3899290d49/docs/trace/extending-the-sdk/Program.cs#L30.

Reply:
Consider taking a look at the other language clients that reached 1.0 stable and see what's the best practice.
One more example from Python https://opentelemetry-python.readthedocs.io/en/latest/getting-started.html.

Reply:
This example was using `TracerContext`. This is how it looks like if we directly use `TracerProvider` to add processors:

```cpp
  auto provider = nostd::shared_ptr<opentelemetry::trace::TracerProvider>(std:move(processor1));
  provider.AddProcessor(std::move(processor2));
  provider.AddProcessor(std::move(processor3));

  opentelemetry::trace::Provider::SetTracerProvider(provider);

```

Reply:
I have modified the example NOT to use `TracerContext`, instead directly use `TracerProvider` to pass processor(s) as an argument. `TracerProvider` constructor takes either of single processor, or vector of processors as an argument.